hack/test_e2e.sh

```diff
@@ -583,15 +583,17 @@ function run_tests {
      # Arguments:                                                              #
      #   arg1: e2e directory                                                   #
      #   arg2: e2e_test, path to test binary                                   #
-     #   arg3: parallel ginkgo tests boolean                                   #
+     #   arg3: ginkgo path to ginko binary                                     #
```

Comment:
I was basically saying just remove this `arg3` and `arg4` from the function and replace it with `${bin_dir}`

Reply:
In that case, we should change arg 2 as well, the path to e2e_test is ${bin_dir}/e2e.test.

I also feel like it time to change the use of "kind" directly and trust the path to ${bin_dir}/kind

Reply:
I'm find with that, we never configure those binaries to be "different" anyways

Reply:
I changed to use ${bin_dir} everywhere needed and also made sure that the binaries in ${bin_dir} is always access through ${bin_dir}/command